tools/mason/MasonArgParse.chpl

```diff
@@ -156,49 +236,72 @@ module MasonArgParse {
           argsList.insert(idx, elems.toArray());
```

Comment:
It's not really about this PR but the above logic changes the argument number -> arg mapping and would complicate error messages along the lines of "Error processing argument 4", if we ever generate those. I'd prefer it if this logic were handled below in the loop processing arguments or perhaps even better, in the appropriate match function(s).

Reply:
> error messages along the lines of "Error processing argument 4", if we ever generate those.

In my survey of other argument parsers, I did not encounter this type of error message detail. I found that most opt to display a message about the specific value or option that had an error. In Chapel specifically, the indices of the original input may not correspond to what the user entered on the CLI. If a user adds any Chapel arguments, or config vars, those values will be consumed before we see the full arguments array. An example can illustrate how indexes may confuse rather than help the user. 

The user enters: `./myProg --myConfigVar 10 --myArgumentName value1 value2 --unknownArgument valueX`
and the program gets `./myProg --myArgumentName value1 value2 --unknownArgument valueX`

In this case, we may wish to tell the user that the argument `--unknownArgument` was not recognized, but our known index would be 4, where the user may perceive it to be 6. I believe that the error message is more useful if it can display the actual value that was not recognized or invalid as opposed to the (potentially incorrect) index value. Even if we choose to add the value and the index, the index offers little additional information that could help the user understand where they went wrong, and may be more confusing than helpful. 

That said, it still may be worth moving the logic that identifies and consumes the `=` to the `_match` method

Reply:
Yeah, those are good points. Subcommands are another setting where the index would differ.

> That said, it still may be worth moving the logic that identifies and consumes the `=` to the `_match` method

That is my feeling on the matter, but I'm not sure I can identify a tangible user-facing benefit. I do think it's a clearer / cleaner code structure, though.

Reply:
I'll add this as a TODO along with other refactoring efforts